src/components/MoneyTemporaryForRefactorRequestConfirmationList.js

```diff
@@ -614,6 +617,221 @@ function MoneyTemporaryForRefactorRequestConfirmationList({
+                        )
+                    }
+                    style={[styles.moneyRequestMenuItem]}
+                    disabled={didConfirm}
+                    interactive={!isReadOnly}
+                    rightLabel={isTagRequired ? translate('common.required') : ''}
+                />
+            ),
+            isRelevant: isTagRequired,
+        },
+        shouldShowTax && {
+            item: (
+                <MenuItemWithTopDescription
+                    key={`${policyTaxRates.name}${taxRateTitle}`}
+                    shouldShowRightIcon={!isReadOnly}
+                    title={taxRateTitle}
+                    description={policyTaxRates.name}
+                    style={[styles.moneyRequestMenuItem]}
+                    titleStyle={styles.flex1}
+                    onPress={() => Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_TAX_RATE.getRoute(iouType, transaction.transactionID, reportID, Navigation.getActiveRouteWithoutParams()))}
+                    disabled={didConfirm}
+                    interactive={!isReadOnly}
+                />
+            ),
+            isRelevant: false,
+        },
+        shouldShowTax && {
+            item: (
+                <MenuItemWithTopDescription
+                    key={`${policyTaxRates.name}${formattedTaxAmount}`}
+                    shouldShowRightIcon={!isReadOnly}
+                    title={formattedTaxAmount}
+                    description={policyTaxRates.name}
+                    style={[styles.moneyRequestMenuItem]}
+                    titleStyle={styles.flex1}
+                    onPress={() => Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_TAX_AMOUNT.getRoute(iouType, transaction.transactionID, reportID, Navigation.getActiveRouteWithoutParams()))}
+                    disabled={didConfirm}
+                    interactive={!isReadOnly}
+                />
+            ),
+            isRelevant: false,
+        },
+        shouldShowBillable && {
+            item: (
+                <View
+                    style={[styles.flexRow, styles.justifyContentBetween, styles.alignItemsCenter, styles.ml5, styles.mr8, styles.optionRow]}
+                    key={translate('common.billable')}
+                >
+                    <Text color={!iouIsBillable ? theme.textSupporting : undefined}>{translate('common.billable')}</Text>
+                    <Switch
+                        accessibilityLabel={translate('common.billable')}
+                        isOn={iouIsBillable}
+                        onToggle={onToggleBillable}
+                    />
+                </View>
+            ),
+            isRelevant: false,
+        },
+    ];
+
+    const relevantFields = _.map(
+        _.filter(classifiedFields, (classifiedField) => !classifiedField || classifiedField.isRelevant),
```

Comment:
Why do we have the condition `!classifiedField`? I don't think we will have any falsey objects in the list of fields, and even if we did, how would that make it "relevant"?

Reply:
This is a great catch and I missed this change. It's actually my fault, as I kind of naively suggested porting the React conventions to "pure" JavaScript, without thinking much of how this will work out.

The root cause is the expression `shouldShowSmartScanFields && { ... }`, which evaluates to `false` if `shouldShowSmartScanFields` is `false`. This is indeed smelly.

Maybe we could go with...

```ts
const classifiedFields = _.filter([
  shouldShowSmartScanFields ? { /* stuff */ } : undefined,
  // more stuff
], classifiedField !== undefined);
```

What do you guys think?

Reply:
Ah ok I see, makes sense. I actually missed the fact that some elements would be `false`, because it's easy to miss that this is "pure" JS.

I think your suggestion is good. I might suggest setting a `shouldRender` field in the item object, instead of using the ternary expression, then filtering to include only items that shouldRender. I think that's slightly more clear. Either way it will be better then its current form.

Reply:
@dukenv0307 This is a good observation! I agree that `shouldShow` is a suitable name here.